ports/unix/mpbthciport.c

```diff
@@ -60,12 +61,23 @@ extern bool mp_bluetooth_hci_poll(void);
 // For synchronous mode, we run all BLE stack code inside a scheduled task.
 // This task is scheduled periodically (every 1ms) by a background thread.
 
+// A signal to use to signal the main thread that a scheduled function is ready.
+// TODO how to choose this signal value?
```

Comment:
This is a tough one. Ultimately, it is up to the "application" to define what signals mean. So if MicroPython was just an application it could be anything. But since it can be an embedded library as well, it should probably be configurable with a sane default in case it conflicts with a signal being used in the application where MicroPython is embedded. Also, MicroPython is a runtime for applications that people write, so users won't be able to use whatever signal is chosen. On Linux it is no so bad since there are plenty of `SIGRTMIN` signals. But macOS doesn't have these, so we are basically left with `SIGUSR1` (used by GC in MicroPython already) and `SIGUSR2`, which leaves nothing leftover for user programs.

Reply:
Yeah, it's tough.  Might need to be more sophisticated and use semaphores.  As I mention above this approach is probably broken so needs more thought.